central/alert/datastore/datastore.go

```diff
@@ -56,9 +58,28 @@ func New(alertStore store.Store, searcher search.Searcher, platformMatcher platf
 }
 
 // GetTestPostgresDataStore provides a datastore connected to postgres for testing purposes.
-func GetTestPostgresDataStore(_ testing.TB, pool postgres.DB) DataStore {
+func GetTestPostgresDataStore(t testing.TB, pool postgres.DB) DataStore {
 	alertStore := pgStore.New(pool)
 	searcher := search.New(alertStore)
+	mockCtrl := gomock.NewController(t)
```

Comment:
I would avoid creating a mock data store here. Most `GetTestPostgresDatastore` functions are run against real db and callers usually expect them to have real datastore. Maybe you can add a `GetTestPostgresDatastore()` to config's datastore, and then use it here?